ui/src/routes/organizations/$orgId/products/$productId/repositories/$repoId/_repo-layout/secrets/index.tsx

```diff
@@ -207,67 +326,185 @@ const RepositorySecrets = () => {
-                <Link
-                  to='/organizations/$orgId/products/$productId/repositories/$repoId/secrets/create-secret'
-                  params={{
-                    orgId: params.orgId,
-                    productId: params.productId,
-                    repoId: params.repoId,
-                  }}
-                >
-                  New secret
-                  <PlusIcon className='h-4 w-4' />
-                </Link>
-              </Button>
-            </TooltipTrigger>
-            <TooltipContent>
-              Create a new secret for this repository
-            </TooltipContent>
-          </Tooltip>
-        </div>
-      </CardHeader>
-      <CardContent>
-        <DataTable
-          table={table}
-          setCurrentPageOptions={(currentPage) => {
-            return {
-              to: Route.to,
-              search: { ...search, page: currentPage },
-            };
-          }}
-          setPageSizeOptions={(size) => {
-            return {
-              to: Route.to,
-              search: { ...search, page: 1, pageSize: size },
-            };
-          }}
-        />
-      </CardContent>
-    </Card>
+    <>
+      <Card>
+        <CardHeader>
+          <CardTitle>Repository Secrets</CardTitle>
+          <CardDescription>Manage secrets for {repo.url}.</CardDescription>
+          <div className='py-2'>
+            <Tooltip>
+              <TooltipTrigger asChild>
+                <Button asChild size='sm' className='ml-auto gap-1'>
+                  <Link
+                    to='/organizations/$orgId/products/$productId/repositories/$repoId/secrets/create-secret'
+                    params={{
+                      orgId: params.orgId,
+                      productId: params.productId,
+                      repoId: params.repoId,
+                    }}
+                  >
+                    New secret
+                    <PlusIcon className='h-4 w-4' />
+                  </Link>
+                </Button>
+              </TooltipTrigger>
+              <TooltipContent>
+                Create a new secret for this repository
+              </TooltipContent>
+            </Tooltip>
+          </div>
+        </CardHeader>
+        <CardContent>
+          <DataTable
+            table={table}
+            setCurrentPageOptions={(currentPage) => {
+              return {
+                to: Route.to,
+                search: { ...search, page: currentPage },
+              };
+            }}
+            setPageSizeOptions={(size) => {
+              return {
+                to: Route.to,
+                search: { ...search, page: 1, pageSize: size },
+              };
+            }}
+          />
+        </CardContent>
+      </Card>
+      <Card className='mt-4'>
+        <CardHeader>
+          <CardTitle>Product Secrets</CardTitle>
+          <CardDescription>
+            Inherited secrets from {product.name} for {repo.url}.
```

Comment:
I would remove "for {repo.url}" (same below).
Also, this commit should close #1926.